include/sound/sof/ext_manifest4.h

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: ((GPL-2.0 WITH Linux-syscall-note) OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ */
+
+#ifndef __SOF_FIRMWARE_EXT_MANIFEST4_H__
+#define __SOF_FIRMWARE_EXT_MANIFEST4_H__
+
+#include <linux/uuid.h>
+
+/* In ASCII  $AE1 */
+#define SOF_EXT_MAN4_MAGIC_NUMBER    0x31454124
+
+#define MAX_MODULE_NAME_LEN     8
+#define MAX_FW_BINARY_NAME      8
+#define DEFAULT_HASH_SHA256_LEN 32
+#define SOF_MAN4_FW_HDR_OFFSET	0x2000
+
+/*********************************************************************
+ *     extended manifest
+ *-------------------
+ *     css_manifest hdr
+ *-------------------
+ *     offset reserved for future
+ *-------------------
+ *     fw_hdr
+ *-------------------
+ *     module_entry[0]
+ *-------------------
+ *     module_entry[1]
+ *-------------------
+ *     ...
+ *-------------------
+ *     module_entry[n]
+ *-------------------
+ *     module_config[0]
+ *-------------------
+ *     module_config[1]
+ *-------------------
+ *     ...
+ *-------------------
+ *     module_config[m]
+ *-------------------
+ *     FW content
+ *-------------------
+ *********************************************************************/
+
+struct sof_ext_manifest4_hdr {
+	u32 id;
+	u32 len; /* length of extension manifest */
+	u16 version_major;
+	u16 version_minor;
+	u32 entries;
+} __packed;
+
+struct sof_man4_fw_binary_header {
+	/* This part must be unchanged to be backward compatible with SPT-LP ROM */
+	uint32_t	id;
+	uint32_t	len; /* sizeof(sof_man4_fw_binary_header) in bytes */
+	uint8_t		name[MAX_FW_BINARY_NAME];
+	uint32_t	preload_page_count; /* number of pages of preloaded image */
+	uint32_t	fw_image_flags;
+	uint32_t	feature_mask;
+	uint16_t	major_version;
+	uint16_t	minor_version;
+	uint16_t	hotfix_version;
+	uint16_t	build_version;
```

Comment:
endianness?

Reply:
@plbossart  Hmm, we don't supported mixed-endian CPU/FW in other interfaces either. E.g. just looking at SOF_ABI, it's parsed the same way. We check magic and then cast the version assuming FW follows CPU endianess. I think we discussed this in the past, but I think this would need a bigger rework to support across the board.